cli/command/service/update.go

```diff
@@ -433,6 +434,41 @@ func getUpdatedSecrets(apiClient client.APIClient, flags *pflag.FlagSet, secrets
 		}
 		secrets = append(secrets, addSecrets...)
 	}
+
+	// If we're being asked to replace a specific secret, let's do in-place replacement of names and IDs
+	if flags.Changed(flagSecretReplace) {
+		secretRefQuery := []*types.SecretRequestOption{}
+
+		// Retrieve all of the secret1=secret2 replacement mappings and create a list of SecretRequestOption
+		replaceValues := flags.Lookup(flagSecretReplace).Value.(*opts.MapOpts).GetAll()
+		for _, value := range replaceValues {
+			secretRefQuery = append(secretRefQuery, &types.SecretRequestOption{Source: value})
+		}
+
+		// Retrieve a list of secret references with the SecretID filled in
+		fullSecretRefs, err := parseSecrets(apiClient, secretRefQuery)
+		if err != nil {
+			return nil, err
+		}
+
+		// Create a map with all of the full secretReferences
```

Comment:
Non-blocking:  since the secret references aren't actually full (there are no target details we care about, for instance), would it make sense to just make a map of secret name to secret ID?